pelican/log.py

```diff
@@ -88,12 +88,16 @@ class FatalLogger(LimitLogger):
     # adding `stacklevel=2` means that the displayed filename and line number
     # will match the "original" calling location, rather than the wrapper here
     def warning(self, *args, **kwargs):
-        super().warning(*args, stacklevel=2, **kwargs)
+        if "stacklevel" not in kwargs.keys():
+            kwargs["stacklevel"] = 2
+        super().warning(*args, **kwargs)
```

Comment:
We can make this "transparent", by effectively doing `stacklevel + 1`, right? `+1` to pass through the `super().warning(...)` part? Otherwise everyone needs to account for it, like you do above with `stacklevel = 3`. And if this is refactored or no longer overridden, everything needs to change.

I feel like there is a better way to do this. Something like:

```python
def warning(self, *args, stacklevel=1, **kwargs):
    kwargs['stacklevel'] = stacklevel + 1
    super().warning(*args, **kwargs)
```

should work. But I didn't test it and it's late so I might not be fully awake :).

Reply:
Done!